src/api.rs

```diff
@@ -5,13 +5,54 @@ use job::{Code, CodeImpl, Job};
 use std::sync::Arc;
 use thread_pool::{self, Registry, WorkerThread};
 
-/// Initializes the Rayon threadpool. You don't normally need to do
-/// this, as it happens automatically, but it is handy for
-/// benchmarking purposes since it avoids initialization overhead in
-/// the actual operations.
-pub fn initialize() {
-    let registry = thread_pool::get_registry();
-    registry.wait_until_primed();
+#[derive(Debug,PartialEq)]
+pub enum InitResult {
+    NumberOfThreadsZero,
+    NumberOfThreadsNotEqual
+}
+
+#[derive(Debug)]
+pub struct Configuration {
+    pub bench: bool,
+    pub num_threads: Option<usize>
+}
+
+impl Configuration {
+    pub fn new() -> Configuration {
+        Configuration { bench: false, num_threads: None }
+    }
+
+    pub fn set_num_threads(mut self, num_threads: usize) -> Configuration {
+        self.num_threads = Some(num_threads);
+        self
+    }
+
+    pub fn set_bench(mut self) -> Configuration {
+        self.bench = true;
+        self
+    }
+
+    pub fn initialize(self) -> Result<(), InitResult> {
+        if let Some(value) = self.num_threads {
+            if value == 0 {
+                return Err(InitResult::NumberOfThreadsZero);
+            }
+        }
+
+        let registry = thread_pool::get_registry(self.num_threads);
+
+        if let Some(value) = self.num_threads {
+            if value != registry.num_threads() {
+                return Err(InitResult::NumberOfThreadsNotEqual);
+            }
+        }
+
+        if self.bench {
```

Comment:
It might be better to just wait unconditionally. It's not going to be a long wait, I expect.